app/views/components_refactored/input_date.scala.html

```diff
@@ -0,0 +1,46 @@
+@*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import uk.gov.hmrc.govukfrontend.views.html.components._
+@import uk.gov.hmrc.hmrcfrontend.views.html.components.implicits._
+@import uk.gov.hmrc.govukfrontend.views.html.components.implicits._
+
+@this( govukDateInput : GovukDateInput )
+
+@(
+    field: Field,
+    legend: String,
+    headingIsLegend: Boolean,
+    legendClass: Option[String] = None,
+    hint: Option[String] = None
+)(implicit messages: Messages)
+
+@govukDateInput(DateInput(
+    id = field.id,
+    namePrefix = None,
+    fieldset = Some(Fieldset(
+        legend = Some(Legend(
+            content = Text(legend),
+            classes = "govuk-fieldset__legend--xl",
+            isPageHeading = headingIsLegend
+        ))
+    )),
+    hint = hint.map(hintValue => Hint(content = Text(hintValue))),
+    errorMessage = field.error match {
```

Comment:
minor: this could written as `field.error.map...` to avoid the `None => None` case

Reply:
this has been updated now.